extensions/wikia/Flags/Flags.hooks.php

```diff
@@ -65,38 +65,60 @@ public static function onPageHeaderDropdownActions( array &$actions ) {
 	}
 
 	/**
-	 * Hooks into the internalParse() process and injects a wikitext
-	 * with notices for the given page.
-	 * @param \Parser $parser
-	 * @param string $text
-	 * @param \StripState $stripState
+	 * Modifies the original ParserOutput object using the one returned from FlagView.
+	 * The modification
+	 * @param \ParserOutput $parserOutput
+	 * @param \Article $article
 	 * @return bool
 	 */
-	public static function onParserBeforeInternalParse( \Parser $parser, &$text, &$stripState ) {
-		/**
-		 * Don't check for flags if:
-		 * - Parser is not used for the main content
-		 * - Flags have already been parsed
-		 * - a user is on an edit page
-		 * - the request is from VE
-		 */
-		$helper = new FlagsHelper();
-		if ( $parser->getOptions()->getIsMain()
-			&& $parser->mFlagsParsed !== true
-			&& $helper->shouldInjectFlags() ) {
-			$addText = ( new \FlagsController )
-				->getFlagsForPageWikitext( $parser->getTitle()->getArticleID() );
-
-			if ( $addText !== null ) {
-				$mwf = \MagicWord::get( 'flags' );
-				if ( $mwf->match( $text ) ) {
-					$text = $mwf->replace( $addText, $text );
-				} else {
-					$text = $addText . $text;
+	public static function onBeforeParserCacheSave( \ParserOutput $parserOutput, \Article $article ) {
```

Comment:
`\Article` should be `\Page`, as that is the type used in the signature of `$article` from `ParserCache::save`, and generally it's better to specify `Page` when dealing with Article and WikiPage objects.